components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/autofill/GeckoLoginDelegateWrapper.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.autofill
+
+import mozilla.components.concept.storage.Login
+import mozilla.components.concept.storage.LoginStorageDelegate
+import org.mozilla.geckoview.GeckoResult
+import org.mozilla.geckoview.LoginStorage
+
+/**
+ * This class exists only to convert incoming [LoginEntry] arguments into [Login]s, then forward
+ * them to [storageDelegate]. This allows us to avoid duplicating [LoginStorageDelegate] code
+ * between different versions of GeckoView, by duplicating this wrapper instead.
+ */
+class GeckoLoginDelegateWrapper(private val storageDelegate: LoginStorageDelegate) : LoginStorage.Delegate {
```

Comment:
This one is currently in nightly.  We will definitely need to triplicate it, but the API isn't available in Gecko beta yet.  This will have to be a follow up.

Reply:
Got it, thanks! Can you file follow-up issues to do this work when GV versions are bumped?

Reply:
👍 
https://github.com/mozilla-mobile/android-components/issues/5530